ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -123,8 +129,9 @@ void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> f
     try {
       GetBlobOperation getBlobOperation =
           new GetBlobOperation(routerConfig, routerMetrics, clusterMap, responseHandler, blobId, futureResult, callback,
-              operationCompleteCallback, blobIdFactory, time);
+              operationCompleteCallback, readyForPollCallback, blobIdFactory, time);
       getOperations.add(getBlobOperation);
+      readyForPollCallback.onPollReady();
```

Comment:
I think the current way is correct. When a new operation is being submitted, the `NetworkClient` is waken up by the operation-submitting thread, which is different from the `OperationController` thread (which may be sleeping already).